src/demikernel/dbuf.rs

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+//==============================================================================
+// Imports
+//==============================================================================
+
+use ::runtime::{
+    fail::Fail,
+    memory::Buffer,
+};
+use ::std::{
+    fmt,
+    fmt::Debug,
+    ops::{
+        Deref,
+        DerefMut,
+    },
+    slice,
+    sync::Arc,
+};
+
+//==============================================================================
+// Structures
+//==============================================================================
+
+/// Data Buffer.
+#[derive(Clone)]
+pub struct DataBuffer {
+    /// Underlying data.
+    data: Option<Arc<[u8]>>,
+
+    /// Data offset.
+    offset: usize,
+
+    /// Data length
+    len: usize,
+}
+
+//==============================================================================
+// Associated Functions
+//==============================================================================
+
+/// Associated Functions for Data Buffers
+impl DataBuffer {
+    // Creates a data buffer with a given capacity.
+    pub fn new(capacity: usize) -> Result<Self, Fail> {
+        // Bad capacity.
```

Comment:
"Check Arguments." would be a better comment here.

Reply:
Sure.